src/rgw/rgw_swift_auth.cc

```diff
@@ -418,21 +420,20 @@ ExternalTokenEngine::authenticate(const std::string& token,
   return result_t::grant(std::move(apl));
 }
 
-static int build_token(const string& swift_user,
-                       const string& key,
-                       const uint64_t nonce,
-                       const utime_t& expiration,
-                       bufferlist& bl)
+static ceph::bufferlist build_token(CephContext* const cct,
+                                    const std::string& swift_user,
+                                    const std::string& key,
+                                    const uint64_t nonce,
+                                    const utime_t& expiration)
 {
+  ceph::bufferlist bl;
   ::encode(swift_user, bl);
   ::encode(nonce, bl);
   ::encode(expiration, bl);
 
   bufferptr p(CEPH_CRYPTO_HMACSHA1_DIGESTSIZE);
 
-  char buf[bl.length() * 2 + 1];
-  buf_to_hex((const unsigned char *)bl.c_str(), bl.length(), buf);
-  dout(20) << "build_token token=" << buf << dendl;
+  ldout(cct, 20) << "build_token token=" << buf_to_hex(bl) << dendl;
```

Comment:
I believe this should be enough:
`ldout(cct, 20) << "build_token token=" << buf_to_hex(bl) << dendl;`
`token` seems redundant here